### PR TITLE
ResStock-HPXML: Expand HPXML Regression Testing

### PR DESCRIPTION
**Pull request by aspeake**
Expands the results csv comparison script to allow comparisons between the original output variables of `SimulationOutpoutReport` and the updated output variables on the `restructure-v3` branch

## Checklist

Not all may apply:

- [ ] Unit tests have been added or updated
- [ ] All rake tasks have been run, and pass
- [ ] Documentation has been modified appropriately
- [ ] Any new options are added to `project_testing`
- [ ] `project_testing` runs without any failures
- [ ] No unexpected regression test changes
- [ ] All tests are passing
- [ ] The [changelog](https://github.com/NREL/resstock/blob/master/CHANGELOG.md) has been updated appropriately
- [ ] This branch is up-to-date with develop

For more information on how to perform these checklist items, see the documentation's [Advanced Tutorial](https://resstock.readthedocs.io/en/latest/advanced_tutorial/index.html).

### CODE REVIEW

File: test/res_results_csv_comparisons.py [hunk 1]
```diff
@@ -290,14 +326,41 @@ def regression_tables(self):
...
+        model_type_map = {'Single-Family Detached':'SFD', 'Single-Family Attached':'SFA', 'Multi-Family':'MF'}
+        deltas['build_existing_model.geometry_building_type_recs'] = deltas['build_existing_model.geometry_building_type_recs'].map(model_type_map)
+        deltas.to_csv(output_path)
 
 if __name__ == '__main__':
 
     # Inputs
-    base_table_name = sys.argv[1]
-    feature_table_name = sys.argv[2]
+    parser = argparse.ArgumentParser()
```
Comment (joseph-robertson) [inline test/res_results_csv_comparisons.py:349]: Good use of argparse.

File: test/res_results_csv_comparisons.py [hunk 2]
```diff
@@ -290,14 +326,41 @@ def regression_tables(self):
...
+    parser.add_argument("base_table_name", help="Filepath to base results table", type=str)
+    parser.add_argument("feature_table_name", help="Filepath to feature results table", type=str)
+    parser.add_argument("--use_cols", help="Which table's variable names to use (int)", type=int)  # Only can map from old to new right now
+    args = parser.parse_args()
```
Comment (joseph-robertson) [inline test/res_results_csv_comparisons.py:353]: Can you include an example usage line?

File: test/res_results_csv_comparisons.py [hunk 3]
```diff
@@ -290,14 +326,41 @@ def regression_tables(self):
...
+
+    base_table_name = args.base_table_name
+    feature_table_name = args.feature_table_name
+    if args.use_cols==1:
+        cols_to_use = 'base'
+    elif args.use_cols==2:
```
Comment (joseph-robertson) [inline test/res_results_csv_comparisons.py:359]: Based on your comment above, this must always be a "2" right now?

**Review by joseph-robertson — CHANGES_REQUESTED** [review]
Consider comparing cost multipliers, like we discussed.

Before merging this into develop, I think we need to remember to delete:
* column_mapping.csv?
* develop_test_run.csv?

File: test/res_results_csv_comparisons.py (hunk 3 above)
Comment (aspeake) [reply test/res_results_csv_comparisons.py:359]: It should always be the table that uses the latest variable names (ie, `restructure-v3`)

File: test/res_results_csv_comparisons.py (hunk 2 above)
Comment (aspeake) [reply test/res_results_csv_comparisons.py:354]: added in https://github.com/NREL/resstock/pull/554/commits/dea74b3970c2ddb05660255439da7e2b822e317b

**Review by joseph-robertson — APPROVED** [review]

**Pull request merged**